#25 Implemented a blacklist for BaseApps

Eonfge opened the pull request:
See issue #25 where I mentioned this minor issue

Eonfge committed:
#25 Implemented a blacklist for BaseApps

tchx84 commented on src/model.js:
```diff
@@ -25,6 +25,21 @@ const _group = 'Context';
+    'org.electronjs.Electron.BaseApp',
+    'org.electronjs.Electron2.BaseApp',
+    'org.godotengine.godot.BaseApp',
+    'org.mozilla.Firefox.BaseApp'
```
I don't think maintaining a hard coded list of app-ids would scale well. But, we could add a `_isBaseApp(appId)` private method to the model though, which could translate to ` return appId.endsWith('.BaseApp')`.

A bit of a hack, but I think is a decent trade-off considering `.BaseApp` is a well established convention at this point. 

EDIT: added `_` to "make" it private.

Eonfge commented:
I first had that, but I wish to argue that it's better to a limited blacklist of hard defined app slugs.

See, there could also be other Flatpak providers then Flathub, and they are in no way obliged to follow the same naming schematic. See [Nuvola](https://nuvola.tiliado.eu/) for example. As such, it's better to cast a net that occasionally has a false negative, then a false positive: Better not block one BaseApp from an unknown third party repository, then to block legitimate apps because string validations are inherently messy.

So, while we're add it, shall we also add `eu.tiliado.NuvolaBase` to the blacklist?

tchx84 commented:
> I first had that, but I wish to argue that it's better to a limited blacklist of hard defined app slugs.
> 
> See, there could also be other Flatpak providers then Flathub, and they are in no way obliged to follow the same naming schematic. See [Nuvola](https://nuvola.tiliado.eu/) for example. As such, it's better to cast a net that occasionally has a false negative, then a false positive: Better not block one BaseApp from an unknown third party repository, then to block legitimate apps because string validations are inherently messy.
> 
> So, while we're add it, shall we also add `eu.tiliado.NuvolaBase` to the blacklist?

I rather see those .BaseApp in the app, which are innocuous, rather than adding the burden of maintaining a hard-coded list. You said it yourself, there could be many providers, and having a hard-coded list is just an endless mouse cat chase game.

To clarify; I am open to the idea of not-showing BaseApps, but only if we can find a way of doing it without creating bigger problems than the issue it intends to solve.

Eonfge committed:
Simplified BaseApp filtering

Will not work with third party repositories, so this should be reconsidered once more repositories emerge.

tchx84 commented on src/model.js:
```diff
@@ -363,7 +347,7 @@ var FlatsealModel = GObject.registerClass({
             const activePath = GLib.build_filenamev([file.get_path(), 'current', 'active']);
 
-            if (GLib.access(activePath, 0) === 0 && _blacklist.includes(appId) == false)
+            if (GLib.access(activePath, 0) === 0 && appId.endsWith(".BaseApp") == false)
```
Can you:
1. Invert the order? Check for `.BaseApp` first. One disk access less ;)
2. Use single quotes.
3. Add a `/* XXX will only work with app-ids that follow the BaseApp convention */` (or whatever message you seem fit).

tchx84 reviewed (changes requested):
Plus, can you add a test for this new behavior? 

Basically, just add something like https://github.com/tchx84/Flatseal/tree/master/tests/content/system/flatpak/app/com.test.Example/current/active to the `tests/content` directory. Call it `com.test.BaseApp` and add a specific test in `tests/src/model.js`  to check that is the app-id is properly ignored.

tchx84 commented:
Also, if you are not convince about this approach either,  you could research a bit more and see if there is another way to detect this condition. But if you're out of time, this will be fine for now.

Eonfge committed:
Added comment related to the implementation scope

Eonfge commented:
Added the comment. Considering the size and relevance of this feature, I think it's good enough for now.

tchx84 commented:
I appreciate your feedback and interest in contributing to the project, but let me be clear; I can't accept contributions that won't consider such basic requirements like adding a simple test for a new behavior.
